### ISSUE TITLE
Checkboxes of optional properties are inconsistent when adding items to an array.

### ISSUE THREAD

**Issue by Henri-A-S** [json-editor/json-editor#1586]
### General information

- json-editor version: 2.15.1

### Expected behavior

Adding an item in an array should always produce the same item. i.e. the checkboxes of optional properties should be consistent.

### Actual behavior

The checkboxes of optional properties depend on the cached item at this position.

### Steps to reproduce the behavior

- Add a first item
- Check the checkbox of the property "type"
- Delete the item
- Add a new item

The first time the checkbox is not set. When adding the item a second time it is set.

Direct link to example: [playground](https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUBXRBTA+vAhgLa75RQBOMARulFsigGb4A22ANCBOWIdDlSyMw5XABMsLLHVTM2WTpnFD86FlBwA3VunqzWHEIXwAPHBOgALVAAZONMghyIopXITATUARk4BjfGxnLCQYWE1cbkgscigAT2D8cj9rJgMFEERLMAB3HBiecgYQGHg6cnw/WAQQTii+DQhAuhxjMyl4AHMoSxxRSsssMX15TlEPCJw5IKiIGNg9NNGQEOFk3D8EFxHDGE34FmpUEGEy/Bz6XiwAVlrwKgArLCqcFnw4sFpj+BFjFjvECksMZUKB4nNjkkKnE7mFgQwwXEISh7k8qndZvMYItEcjMhRSp0QABfUmcHrArDHKhgMBQFwVCAAFjuogAjugYKIxAIEhJmGooDsMmIYIgIG8EuzOdycAgWDCloYsrk5fxSqgKLpOD8cPgxKLqgQWDhMbFsQxphl8A9TMLOKLEPgqFJ8oacA9EDUlSKxc7XZsWG8INh7SBHf7Ijw5ubFlaHX6XSRyNC9QawxGk3qU+8+lgRBJyBnE66obmJFIZD7OCFI9nU5sIIr4yAywl9hQwGwcFBIMWnVm2+ZJNISEG+rlLekEwPSzm+SOWm8XBOcsLSUA=)

```javascript
{
  "type": "array",
  "items": {
    "type": "object",
    "properties": {
      "type": "string"
    }
  }
}
```

**Comment by ThomasKiesel** [comment]
Is this issue still open? It seems to me that the bug has been fixed with the latest merge.

**Comment by schmunk42** [comment]
Commit is available in 2.15.2

https://github.com/json-editor/json-editor/commit/deca8111037d718cd7443d58ba6642451cb93c6d

**Comment by schmunk42** [comment]
@germanbisurgi is it OK to close this?
